zone/lua_client.cpp

```diff
@@ -20,6 +20,10 @@
 
 struct InventoryWhere { };
 
+struct Lua_Mob_List {
+	std::vector<Lua_Mob> entries;
```

Comment:
I think using the luabind stl iterators for new stuff should be avoided when possible and just a table returned. Their only use case looks like returning access to containers that already exists on exported classes. This (and entity_list usages) just construct and copy everything to a new vector anyway for no benefit. Luabind also ends up making another copy of the vector when it copies the exported class instance for ownership.

I guess an argument for using them could be that they match current entity_list apis but I still think just returning a table would be better.

Reply:
Not sure that I understand fully what you’re suggesting. 

The goal is to just return a list of mobs to be able to iterate through in a loop like any other entity list call. 

Can you give Kingly (and myself) an example of an alternative, why it’d be better and how it’d look with usage

Reply:
I'm suggesting changing the return to a lua table instead of a Lua_Mob_List that wraps a vector. Then it can just be iterated in a conventional lua way:
```lua
local party = client:GetGroupOrRaidOrSelf()
eq.debug("party count: " .. #party)
for i, mob in pairs(party)
  -- ...
end
```

see https://github.com/EQEmu/Server/blob/82aa6a1587477e642e7ac44e7902f4568bb8563e/zone/lua_mob.cpp#L3307